fahrschein/src/test/java/org/zalando/fahrschein/NakadiReaderTest.java

```diff
@@ -606,4 +608,29 @@ public void shouldCloseOnRuntimeException() throws IOException {
         }
     }
 
+    @Test
+    public void shouldCloseOnInactiveListener() throws IOException {
+        final ClientHttpResponse response = mock(ClientHttpResponse.class);
+        final ByteArrayInputStream inputStream = new ByteArrayInputStream("{\"cursor\":{\"partition\":\"123\",\"offset\":\"456\"},\"foo\":\"bar\",\"events\":[{\"id\":\"789\"}],\"metadata\":{\"foo\":\"bar\"}}".getBytes("utf-8"));
+        when(response.getBody()).thenReturn(inputStream);
+
+        final ClientHttpRequest request = mock(ClientHttpRequest.class);
+        when(request.execute()).thenReturn(response);
+        final HttpHeaders headers = new HttpHeaders();
+        when(request.getHeaders()).thenReturn(headers);
+
+        when(clientHttpRequestFactory.createRequest(uri, HttpMethod.GET)).thenReturn(request);
+
+        final NoBackoffStrategy backoffStrategy = new NoBackoffStrategy();
+
+        ReaderManager readerManager = (Optional<Subscription> subscription) -> false;
```

Comment:
That's true.

I provide now two test case. one to verify that the reading discontinues when the readermanager flags not to continue reading. the other one, that checks if that the reading is not started at all, if the reader disallows it